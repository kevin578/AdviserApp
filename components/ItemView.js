import React, { Component } from "react";
import { View, Text, ScrollView, StyleSheet, Dimensions } from "react-native";
import { connect } from "react-redux";
import HTML from "react-native-render-html";

const styles = StyleSheet.create({
  container: {
    padding: 10
  },
  p: {
    backgroundColor: "green",
    padding: 0
  },
  h2: {
    color: "#FF3366" // make links coloured pink
  }
});

class ItemView extends Component {
  render() {
    const { props } = this;
    return (
      <ScrollView style={styles.container}>
        <HTML
          html={props.html}
          imagesMaxWidth={Dimensions.get("window").width - 20}
        />
      </ScrollView>
    );
  }
}

function mapStateToProps(state) {
  return {
    html: state.application.itemHtml
  };
}

export default connect(mapStateToProps)(ItemView);
