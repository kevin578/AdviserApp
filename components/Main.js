import React, { Component } from "react";
import { View, Text } from "react-native";

const styles = {
  body: {
    padding: 60
  }
};

export default class Main extends Component {
  render() {
    return (
      <View style={styles.body}>
        <Text>Here is the main page.</Text>
      </View>
    );
  }
}
