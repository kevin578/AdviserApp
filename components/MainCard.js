import React, { Component } from "react";
import { View, Text } from "react-native";

const styles = {
  container: {
    width: "90%",
    backgroundColor: "#fff",
    borderRadius: 2,
    marginLeft: "auto",
    marginRight: "auto", 
    marginBottom: 20,
    padding: 20
  },
  titleText: {
      fontSize: 18,
      fontWeight: "600",
      marginBottom: 10
  }
};

export default class MainCard extends Component {
  render() {
    const { props } = this;
    return (
      <View style={styles.container}>
        <Text style={styles.titleText}>{props.title}</Text>
        <Text>{props.excerpt}</Text>
      </View>
    );
  }
}
