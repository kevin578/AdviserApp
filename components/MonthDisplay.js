import React, { Component } from "react";
import { View, Text, TouchableHighlight, StyleSheet } from "react-native";
import {connect} from "react-redux";
import { Icon } from "react-native-elements";
import Touchable from "react-native-platform-touchable";
import * as actions from "../actions";

const monthContainerHeight = 100;

const styles = StyleSheet.create({
  MonthDisplayContainer: {
    marginTop: 0,
    flexDirection: "row",
    justifyContent: "space-around",
    bottom: monthContainerHeight / 2
  },
  monthContainer: {
    width: 130,
    justifyContent: "center",
    alignItems: "center",
    height: monthContainerHeight,
    backgroundColor: "#fff",
    borderRadius: 50,
    borderWidth: 3,
    borderColor: "#212121"
  },

  arrowContainer: {
    top: 25,
    borderWidth: 3,
    borderColor: "#212121",
    height: 50,
    width: 50,
    borderRadius: 30,
    backgroundColor: "#fff",
    justifyContent: "center",
    alignItems: "center"
  }
});

class MonthDisplay extends Component {
  touchButton = () => {
    console.log("clicked");
  };

  render() {
      const { month } = this.props;
    return (
      <View style={styles.MonthDisplayContainer}>
        <Touchable onPress={this.touchButton}>
          <View style={styles.arrowContainer}>
            <Icon name="arrow-back" />
          </View>
        </Touchable>
        <View style={styles.monthContainer}>
          <Text style={styles.monthText}>{month.currentMonth}</Text>
        </View>
        <Touchable>
          <View style={styles.arrowContainer}>
            <Icon name="arrow-forward" />
          </View>
        </Touchable>
      </View>
    );
  }
}

function mapStateToProps(state) {
    return ({
      month: state.month
    })
  }
  
  export default connect(mapStateToProps, actions)(MonthDisplay);