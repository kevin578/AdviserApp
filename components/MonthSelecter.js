import React, { Component } from "react";
import { connect } from "react-redux";
import { StyleSheet, View, Text, Dimensions } from "react-native";
import { Icon } from "react-native-elements";
import Touchable from "react-native-platform-touchable";
import Carousel from "react-native-snap-carousel";
import * as actions from "../actions";
import { months } from "../monthsArray";
import SideMenu from "./SideMenu";

const monthContainerHeight = 100;

const styles = StyleSheet.create({
  container: {
    width: Dimensions.get("window").width,
    height: 150
  },
  monthCircle: {
    marginLeft: "auto",
    marginRight: "auto",
    marginTop: 75,
    width: 120,
    height: 75
  },

  monthText: {
    fontSize: 18,
    fontWeight: "600"
  },

  yearText: {
    fontSize: 12,
    paddingTop: 5
  },

  MonthDisplayContainer: {
    flexDirection: "row",
    justifyContent: "space-around",
    bottom: monthContainerHeight / 2
  },
  monthContainer: {
    width: 130,
    justifyContent: "center",
    alignItems: "center",
    height: monthContainerHeight,
    backgroundColor: "#fff",
    borderRadius: 50,
    borderWidth: 3,
    borderColor: "#212121"
  },

  arrowContainer: {
    top: 25,
    borderWidth: 3,
    borderColor: "#212121",
    height: 50,
    width: 50,
    borderRadius: 30,
    backgroundColor: "#fff",
    justifyContent: "center",
    alignItems: "center"
  },

  hideArrowContainer: {
    opacity: 0
  }
});

class MonthSelecter extends Component {
  state = {
    width: 375
  };

  componentDidMount() {
    const { props } = this;
    const { width } = Dimensions.get("window");
    this.setState({ width });
    if (!props.month.currentMonth) {
      const currentDate = new Date();
      const month = this.convertMonth(currentDate.getMonth());
      props.setCurrentMonth(month);
    }

  }

  convertMonth = realMonth => {
    const appMonths = [null, 6, 7, 8, 9, 10, 11, 12, 1, 2, 3, 4, 5];
    return appMonths[realMonth];
  };

  renderItem = ({ item, index }) => {
    const { name, backgroundColor } = item;
    return (
      <View
        style={StyleSheet.flatten([styles.container, { backgroundColor }])}
      />
    );
  };

  getArrowStyle = arrow => {
    const { month } = this.props;
    const { currentMonth } = month;
    if (arrow === "left" && currentMonth < 1) {
      return StyleSheet.flatten([styles.hideArrowContainer]);
    }
    if (arrow === "right" && currentMonth > 10) {
      return StyleSheet.flatten([styles.hideArrowContainer]);
    }
    return {};
  };

  onScroll = event => {
    const { setCurrentMonth } = this.props;
    setCurrentMonth(event);
  };

  nextMonth = () => {
    const { month, setCurrentMonth } = this.props;
    const { currentMonth } = month;
    if (currentMonth < 11) {
      setCurrentMonth(currentMonth + 1);
      this.carousel.snapToNext();
    }
  };

  prevMonth = () => {
    const { month, setCurrentMonth } = this.props;
    const { currentMonth } = month;
    if (currentMonth > 0) {
      setCurrentMonth(currentMonth - 1);
      this.carousel.snapToPrev();
    }
  };

  render() {
    const { month, user } = this.props;
    const { currentMonth } = month;
    const { width } = this.state;
    return (
      <View>
        <Carousel
          ref={c => {
            this.carousel = c;
          }}
          data={months}
          onSnapToItem={this.onScroll}
          renderItem={this.renderItem}
          sliderWidth={width}
          itemWidth={width}
          firstItem={currentMonth}
        />

        <View style={styles.MonthDisplayContainer}>
          <Touchable
            onPress={this.prevMonth}
            style={this.getArrowStyle("left")}
          >
            <View style={styles.arrowContainer}>
              <Icon name="arrow-back" />
            </View>
          </Touchable>
          <View style={styles.monthContainer}>
            <Text style={styles.monthText}>{months[currentMonth].name}</Text>
            <Text style={styles.yearText}>{user.year} Year</Text>
          </View>
          <Touchable
            onPress={this.nextMonth}
            style={this.getArrowStyle("right")}
          >
            <View style={styles.arrowContainer}>
              <Icon name="arrow-forward" />
            </View>
          </Touchable>
        </View>
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    month: state.month,
    user: state.user
  };
}

export default connect(
  mapStateToProps,
  actions
)(MonthSelecter);
