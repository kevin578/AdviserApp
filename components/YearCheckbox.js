import React, { Component } from "react";
import { connect } from "react-redux";
import { CheckBox } from "react-native-elements";
import { StyleSheet } from "react-native";
import * as actions from "../actions";

const styles = StyleSheet.create({
  checkboxContainerStyle: {
    backgroundColor: "transparent",
    borderWidth: 0,
    paddingLeft: 20,
  },
  checkboxTextStyle: {
    fontSize: 18
  }
});

class YearCheckbox extends Component {

  checkboxPress = year => {
    const { user, clearUserFormYear, setUserFormYear } = this.props;
    const existingYear = user.year;
    if (existingYear === year) {
      clearUserFormYear();
    } else {
      setUserFormYear(year);
    }
  };

  render() {
    const { user, year } = this.props;

    return (
      <CheckBox
        title={year}
        checkedColor="#337ab7"
        checked={user.formYear === year}
        checkedIcon="dot-circle-o"
        uncheckedIcon="circle-o"
        onPress={() => this.checkboxPress(year)}
        containerStyle={styles.checkboxContainerStyle}
        textStyle={styles.checkboxTextStyle}
      />
    );
  }
}
function mapStateToProps(state) {
  return {
    user: state.user
  };
}
export default connect(
  mapStateToProps,
  actions
)(YearCheckbox);
