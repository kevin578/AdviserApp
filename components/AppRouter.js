import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { createStore } from "redux";
import { Scene, Router, Drawer } from "react-native-router-flux";
import YearForm from "./YearForm";
import hamburgerIcon from "../assets/hamburger.png";
import Main from "./Main";
import SideMenu from "./SideMenu";

const styles = StyleSheet.create({
  container: {},
  menuButton: {
    width: 25,
    height: 25
  }
});

export default class AppRouter extends React.Component {
  render() {
    return (
      <Router>
        <Scene key="root" hideNavBar>
          <Scene key="StartBucket" initial>
            <Scene key="Year" component={YearForm} Title="Year" />
          </Scene>

          <Scene
            key="MainBucket"
            drawer
            drawerImage={hamburgerIcon}
            contentComponent={SideMenu}
            style={styles.drawer}
          >
            <Scene
              key="Main"
              component={Main}
              title="Adviser App"
              leftButtonIconStyle={styles.menuButton}
            />
            <Scene
              key="changeYear"
              component={YearForm}
              title="Adviser App"
              leftButtonIconStyle={styles.menuButton}
              updateTitle
            />
          </Scene>
        </Scene>
      </Router>
    );
  }
}
