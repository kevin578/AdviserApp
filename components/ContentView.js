import React, { Component } from "react";
import { Text, View, FlatList } from "react-native";
import axios from "axios";
import { connect } from "react-redux";
import _ from "lodash";
import * as actions from "../actions";
import MainCard from "./MainCard";

const months = [
  "August",
  "September",
  "October",
  "November",
  "December",
  "January",
  "February",
  "March",
  "April",
  "May",
  "June",
  "July"
];

const styles = {
  container: {
    paddingBottom: 250,
    position: "relative",
    bottom: 20
  }
};

class ContentView extends Component {
  componentDidMount() {
    this.loadContent();
  }

  loadContent = async () => {
    const { props } = this;
    try {
      const content = await this.getData(1);
      let posts = content.data;
      const totalPages = content.headers["x-wp-totalpages"];
      if (totalPages > 1) {
        for (let i = 2; i <= totalPages; i += 1) {
          const newContent = this.getData(i);

          posts = posts.concat(newContent.data);
        }
      }
      props.loadContent(await Promise.all(posts));
    } catch (err) {
      return err;
    }
  };

  getData = pageNumber =>
    axios.get("http://adviserapp.local/wp-json/wp/v2/posts", {
      params: {
        page: pageNumber,
        per_page: 99
      }
    });

  filterContent = () => {
    const { props } = this;
    const filteredForMonth = _.filter(props.application.content, {
      content_month: months[props.month.currentMonth]
    });
    const filteredForYears = _.filter(filteredForMonth, "student_year");
    const filteredContent = _.filter(filteredForYears, o =>
      o.student_year.includes(props.user.year)
    );
    return filteredContent;
  };

  renderItem = item => {
    const { title, excerpt, content } = item.item;
    return (
      <MainCard
        title={title.rendered}
        excerpt={excerpt.rendered.slice(3, 140)}
        html={content.rendered}
      />
    );
  };

  render() {
    this.filterContent();
    const { props } = this;
    return (
      <View style={styles.container}>
        <FlatList
          data={this.filterContent()}
          renderItem={this.renderItem}
          keyExtractor={(item, index) => index.toString()}
        />
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    application: state.application,
    month: state.month,
    user: state.user
  };
}

export default connect(
  mapStateToProps,
  actions
)(ContentView);
