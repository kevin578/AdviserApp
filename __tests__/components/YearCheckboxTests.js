import React from "react";
import renderer from "react-test-renderer";
import { YearCheckbox } from "../../components/YearCheckbox";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <YearCheckbox
        year="Freshman"
        user={{
          year: "Freshman"
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
