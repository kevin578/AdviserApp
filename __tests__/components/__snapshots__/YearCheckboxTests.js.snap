// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 0,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 20,
      "opacity": 1,
      "padding": 10,
      "paddingLeft": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        false,
        false,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      onLongPress={undefined}
      onPress={undefined}
      style={
        Array [
          Object {
            "color": "blue",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï†’
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#43484d",
              "fontWeight": "bold",
              "marginLeft": 10,
              "marginRight": 10,
            },
            Object {
              "fontSize": 18,
            },
            undefined,
          ],
        ]
      }
    >
      Freshman
    </Text>
  </View>
</View>
`;
